libcudacxx/include/cuda/std/__concepts/concept_macros.h

```diff
@@ -50,51 +50,46 @@ using __cccl_enable_if_t = typename __cccl_select<_Bp>::template type<_Tp>;
 template <class _Tp, bool _Bp>
 using __cccl_requires_t = typename __cccl_select<_Bp>::template type<_Tp>;
 
-#if (defined(__cpp_concepts) && _CCCL_STD_VER >= 2020)
+#if (defined(__cpp_concepts) && _CCCL_STD_VER >= 2020) || defined(_CCCL_DOXYGEN_INVOKED)
 #  define _CCCL_TEMPLATE(...)               template <__VA_ARGS__>
 #  define _CCCL_REQUIRES(...)               requires __VA_ARGS__
 #  define _CCCL_AND                         &&
 #  define _CCCL_TRAILING_REQUIRES_AUX_(...) requires __VA_ARGS__
 #  define _CCCL_TRAILING_REQUIRES(...)      ->__VA_ARGS__ _CCCL_TRAILING_REQUIRES_AUX_
 #else // ^^^ __cpp_concepts ^^^ / vvv !__cpp_concepts vvv
 #  define _CCCL_TEMPLATE(...)               template <__VA_ARGS__
-#  define _CCCL_REQUIRES(...)               , bool _CCCL_true_ = true, __cccl_enable_if_t < __VA_ARGS__ && _CCCL_true_, int > = 0 >
-#  define _CCCL_AND                         &&_CCCL_true_, int > = 0, __cccl_enable_if_t <
+#  define _CCCL_REQUIRES(...)               , bool __cccl_true_ = true, __cccl_enable_if_t < __VA_ARGS__ && __cccl_true_, int > = 0 >
+#  define _CCCL_AND                         &&__cccl_true_, int > = 0, __cccl_enable_if_t <
```

Comment:
Rather than using an bool, could we do
```suggestion
#  define _CCCL_REQUIRES(...)               , size_t __cccl_line_ = __LINE__, __cccl_enable_if_t < __VA_ARGS__ && __cccl_line_, int > = 0 >
#  define _CCCL_AND                         &&__cccl_line_, int > = 0, __cccl_enable_if_t <
```

Reply:
the types would be the same tho, and compilers would likely error about the default being different. what i was doing with the array types was to make the template parameters different to force the compiler to treat them as separate overloads. i think that's still a good change. i'll try it and see what breaks.